dspace-api/src/main/java/org/dspace/util/SolrUpgradeStatistics6.java

```diff
@@ -0,0 +1,371 @@
+                                        } else if (col == FIELD.scopeId) {
+                                                Object otype = input.getFieldValue("scopeType");
+                                                if (otype != null) {
+                                                        int type = Integer.parseInt(otype.toString());
+                                                        uuid = mapType(type, legacy);
+                                                }
+                                        } else {
+                                                uuid = mapId(col, legacy);
+                                        }
+                                        if (uuid != null) {
+                                                newvals.add(uuid);
+                                        }
+                                } catch (NumberFormatException e) {
+                                        log.warn("Non numeric legacy id "+ col.name() +":" + oval.toString());
+                                }
+                        }
+                        if (newvals.size() > 0) {
+                                input.removeField(col.name());
+                                for(UUID uuid: newvals) {
+                                        input.addField(col.name(), uuid.toString());
+                                }
+                                
+                        }
+                }
+        }
+        
+        private UUID mapId(FIELD col, int val) throws SQLException {
+                
+                if (col == FIELD.owningComm) {
+                        Community comm = communityService.findByLegacyId(context, val);
+                        return comm == null ? null : comm.getID();
+                }
+                if (col == FIELD.owningColl) {
+                        org.dspace.content.Collection coll = collectionService.findByLegacyId(context, val);
+                        return coll == null ? null : coll.getID();
+                }
+                if (col == FIELD.owningItem) {
+                        Item item = itemService.findByLegacyId(context, val);
+                        return item == null ? null : item.getID();
+                }
+                return null;
+        }
+        private UUID mapType(int type, int val) throws SQLException {
+                if (type == Constants.COMMUNITY) {
+                        Community comm = communityService.findByLegacyId(context, val);
+                        return comm == null ? null : comm.getID();
+                }
+                if (type == Constants.COLLECTION) {
+                        org.dspace.content.Collection coll = collectionService.findByLegacyId(context, val);
+                        return coll == null ? null : coll.getID();
+                }
+                if (type == Constants.ITEM) {
+                        Item item = itemService.findByLegacyId(context, val);
```

Comment:
I am going to add some logic to see if item != lastItem and then perform the uncache.